blade-graphics/src/vulkan/resource.rs

```diff
@@ -208,7 +318,17 @@ impl crate::traits::ResourceDevice for super::Context {
                 Buf::SHADER_DEVICE_ADDRESS | Buf::ACCELERATION_STRUCTURE_BUILD_INPUT_READ_ONLY_KHR;
         }
 
-        let raw = unsafe { self.device.core.create_buffer(&vk_info, None).unwrap() };
+        let raw = unsafe {
+            if let Some(mut e) = external_next {
+                self.device
+                    .core
+                    .create_buffer(&vk_info.push_next(&mut e), None)
```

Comment:
let's only do the `push_next` thing inside the `if`, and have `create_buffer` call one and only